webrender/src/render_task.rs

```diff
@@ -458,6 +470,8 @@ impl RenderTask {
             }
         }
 
+        render_task_sanity_check(&outer_rect.size);
```

Comment:
would it be easier (and more robust) if we just had a constructor like `RenderTask::new_dynamic(size, kind, clear_mode, children)` where the check is done?

Reply:
I have a mild preference towards keeping the struct initialization syntax instead of using `::new` methods when construction is simple because the latter isn't very easy to read due to the lack of named arguments.
My hope is that within a few weeks of this sanity check being in the wild we'll have data pointing us to the missing places where we need to be careful about render task sizes and we can then remove the check (or replace it with something a bit better where we check against the real maximum size or something that makes sense on a per-type-of-item basis).

Reply:
> I have a mild preference towards keeping the struct initialization syntax instead of using ::new methods when construction is simple

I agree! In my book the reasoning no longer stands if there is any logic that needs to be done though, which is the case here.

> and we can then remove the check

I don't see why we'd want to remove the check ever. Better put it once and forever in a nice spot to protect us against eternal evils of infinite render tasks.

Reply:
My own book has it that it's not worth losing the readability for a single sanity check function call that is only there for debugging purposes, but that's not really worth arguing over, I'll add the constructor.

Reply:
I don't understand where "losing the readability" comes from. What am I missing?

Reply:
oh, I was misreading it as "reliability", silly me